sources/light_client.move

```diff
@@ -523,3 +522,44 @@ public fun retarget_algorithm(p: &Params, previous_target: u256, first_timestamp
 
     next_target
 }
+
+
+/// Return the amount to `receiver_address` and the OP_RETURN message.
+/// If OP_RETURN is not included in the transaction, return an empty vector.
+/// * `height`: block heigh transacion belong
+/// * `proof`: merkle tree proof, this is the vector of 32bytes
+/// * `tx_index`: index of transaction in block
+/// * `transaction`: bitcoin transaction. Check transaction.move
+/// * `receiver_address`: address of receiver in p2pkh or p2wpkh
+/// @return address and amount for each output
+public fun prove_payment(
```

Comment:
@sczembor - if we are going to use `amount` in nBTC, then we need to know what amount has correct recipient. UTXOs with incorrect recipients must go to a fallback deposit.

Reply:
I think we should return just the amount and tx_id with matched outputs (by receiver address), rather than matching by existence of `op_return`. Then the caller should break down the outputs and amounts based on his knowledge. So, I propose to return: `amount` (at it is currently computed) and `tx_id` only. Optionally, we can return outputs for recipient if needed.

Reply:
In one transaction, only have one OP Return. The output help us know information user want to delivery to SUI.

Reply:
https://learnmeabitcoin.com/technical/script/return/#:~:text=A%20transaction%20can%20only%20include,as%20a%20%22nulldata%22%20script.